### PR TITLE
Pass user locale as suomi.fi auth request param

### PR DESCRIPTION
**Pull request by Gekkio**

### CODE REVIEW

**Review by mikkopiu — APPROVED** [review]
LGTM but did you find any official documentation on this? I managed to only find some random repos with the implementation for `locale`.

File: apigw/src/shared/routes/auth/saml/index.ts [hunk 1]
```diff
@@ -62,8 +63,13 @@ function createLoginHandler({
       req,
       'Login endpoint called'
     )
+    const locale = req.query.locale
+    const options: AuthenticateOptions = {
+      additionalParams: typeof locale === 'string' ? { locale } : {}
```
Comment (mikkopiu) [inline apigw/src/shared/routes/auth/saml/index.ts:68]: More of a Q: will an incorrect locale trigger an error in `passport.authenticate` or will it fail silently? Just wondering whether we should explicitly catch that or not.

**Comment by Gekkio** [comment]
https://palveluhallinta.suomi.fi/fi/tuki/artikkelit/59116c3014bbb10001966f70

> Taulukko 4. Käyttöliittymän kieli, tunnistuspyyntö.

File: apigw/src/shared/routes/auth/saml/index.ts (hunk 1 above)
Comment (Gekkio) [reply apigw/src/shared/routes/auth/saml/index.ts:68]: A non-string locale (= not present, or uses one of the exotic query parameter formats supported by express) -> skipped
A locale unsupported by suomi.fi -> ???

**Pull request merged**